percona-xtradb-cluster/5.6/source/release-notes/Percona-XtraDB-Cluster-5.6.24-25.11.rst

```diff
@@ -0,0 +1,54 @@
+.. rn:: 5.6.24-25.11
+
+=======================================
+ |Percona XtraDB Cluster| 5.6.24-25.11 
+=======================================
+
+Percona is glad to announce the release of |Percona XtraDB Cluster| 5.6 on June 3rd 2015. Binaries are available from `downloads area <http://www.percona.com/downloads/Percona-XtraDB-Cluster-56/release-5.6.24-25.11/>`_ or from our :doc:`software repositories </installation>`.
+
+Based on `Percona Server 5.6.24-72.2 <http://www.percona.com/doc/percona-server/5.6/release-notes/Percona-Server-5.6.24-72.2.html>`_ including all the bug fixes in it, *Galera Replicator* 3.11 and on *Codership wsrep API* 25.11 is now the current **General Availability** release. All of |Percona|'s software is open-source and free, all the details of the release can be found in the `5.6.24-25.11 milestone <https://launchpad.net/percona-xtradb-cluster/+milestone/5.6.24-25.11>`_ at Launchpad.
+
+New Features
+============
+
+ |Percona XtraDB Cluster| now allows reads in non-primary state by introducing a new session variable, :variable:`wsrep_dirty_reads`. This variable is boolean and is ``OFF`` by default. When set to ``ON``, a |Percona XtraDB Cluster| node accepts queries that only read, but not modify data even if the node is in the non-PRIM state (:bug:`1407770`, :bug:`1409618`).
+
+ :variable:`wsrep_replicate_myisam` variable is now a session variable (:bug:`1280280`).
+
+ |Percona XtraDB Cluster| now uses ``getifaddrs`` for node address detection (:bug:`1252700`). 
+
+ |Percona XtraDB Cluster| has implemented two new status variables: :variable:`wsrep_cert_bucket_count` and :variable:`wsrep_gcache_pool_size` for better instrumentation of galera memory usage. :variable:`wsrep_cert_bucket_count` shows the number of cells in the certification index hash-table and :variable:`wsrep_gcache_pool_size` shows the size of the page pool and/or dynamic memory allocated for gcache (in bytes).
+
+Bugs fixed 
+==========
+
+ ``REPLACE``, ``LOAD DATA REPLACE`` and ``INSERT ON DUPLICATE KEY UPDATE`` set exclusive record locks. So when a record was deleted by purge, all record locks acquired by in-progress concurrent statements of those types were lost which could lead to a UK constraint violation. Bug fixed :bug:`1308016`.
```

Comment:
Using concurrent ``REPLACE``, ``LOAD DATA REPLACE`` or ``INSERT ON DUPLICATE KEY UPDATE`` statements in the ``READ COMMITTED`` isolation level or with the ``innodb_locks_unsafe_for_binlog`` option enabled could lead to a unique-key constraint violation.